ansibullbot/triagers/plugins/docs_info.py

```diff
@@ -0,0 +1,306 @@
+import ast
+import dataclasses
+from itertools import filterfalse
+import logging
+import re
+
+import requests
+
+RE_DOCS_PATTERNS = [
+    r"^docs/.*",
+    r"^examples/.*",
+]
+
+RE_DIFF_PATTERNS = {
+    "header": r"^\@+\s+(?P<del_start>[\-]\d+),\d+\s(?P<add_start>[\+]\d+),\d+",
+}
+
+@dataclasses.dataclass(eq=False, order=False)
+class ParsedFunc():
+    name: str
+    line_start: int
+    line_end: int
+    doc_string: str = ""
+    ds_line_start: int = 0
+    ds_line_end: int = 0
+
+@dataclasses.dataclass(eq=False, order=False)
+class ParsedClass():
+    name: str
+    line_start: int
+    line_end: int
+    doc_string: str = ""
+    ds_line_start: int = 0
+    ds_line_end: int = 0
+    funcs: list = dataclasses.field(default_factory=list)
+
+    def find_function(self, lineno):
+        """ Returns a function that contains the given line number ``lineno``. """
+        for item in self.funcs:
+            if item.line_start <= lineno <= item.line_end:
+                return item
+
+@dataclasses.dataclass(eq=False, order=False)
+class ParsedModule():
+    doc_string: str = ""
+    ds_line_start: int = 0
+    ds_line_end: int = 0
+    example_string: str = ""
+    ex_line_start: int = 0
+    ex_line_end: int = 0
+    classes: list = dataclasses.field(default_factory=list)
+
+    def find_class(self, lineno):
+        """ Returns a class that contains the given line number ``lineno``. """
+        for item in self.classes:
+            if item.line_start <= lineno <= item.line_end:
+                return item
+
+class CommitFile:
+    def __init__(self, raw_data):
+        self.raw_data = raw_data
+
+    @property
+    def filename(self):
+        return self.raw_data.get("filename")
+
+    @property
+    def status(self):
+        return self.raw_data.get("status")
+
+    @property
+    def patch(self):
+        return self.raw_data.get("patch")
+
+    @property
+    def raw_url(self):
+        return self.raw_data.get("raw_url")
+
+    @property
+    def file_content(self):
+        if self.raw_url:
+            result = requests.get(self.raw_url)
+            if result.ok:
+                return result.text
+
+def _is_docs_path(filename):
+    """ Determine if affected file is only applicable to documentation directories """
+    for pattern in RE_DOCS_PATTERNS:
+        if re.search(pattern, filename):
```

Comment:
Should we use here just `.startswith` instead of regex?

Reply:
Using `.startswith` is fine by me. The regex approach is just a pre-optimization for cases that are more complex than just the root path.